packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -918,13 +939,85 @@ describe('PhishingDetector', () => {
           ],
           async ({ detector }) => {
-            const { result, type } = detector.check('blocked.com.');
+            const { result, type } = detector.check(
+              formatHostnameToUrl('blocked.com.'),
+            );
 
             expect(result).toBe(true);
             expect(type).toBe('blocklist');
           },
         );
       });
+
+      it('blocks ipfs cid across various formats (cids located in subdomains and paths)', async () => {
+        // CID should not blocked
+        await withPhishingDetector(
+          [
+            {
+              allowlist: [],
+              blocklist: [
+                'QmUDBVyGwqKdSayk7kDKUaj9J41Ft1DWizcKUx5UmgMgGy',
+                'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m',
+                'example.com',
+              ],
+              fuzzylist: [],
+              name: 'first',
+              tolerance: 2,
+              version: 1,
+            },
+          ],
+          async ({ detector }) => {
+            const { result, type } = detector.check(
+              formatHostnameToUrl(
+                'cf-ipfs.com/ipfs/bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze',
+              ),
+            );
+
+            expect(result).toBe(false);
+            expect(type).toBe('all');
+          },
+        );
+
+        // Gateways differ on where the CID is... sometimes in the path, sometimes in a magic subdomain
+        const expectedToBeBlocked = [
+          'ipfs.io/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'gateway.pinata.cloud/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'cloudflare-ipfs.com/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'ipfs.eth.aragon.network/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.dweb.link/#x-ipfs-companion-no-redirect',
+          'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.cf-ipfs.com/#x-ipfs-companion-no-redirect',
+          'example.com',
+          'example.com/foo/bar',
+        ];
+
+        // CID should be blocked
+        for await (const entry of expectedToBeBlocked) {
+          await withPhishingDetector(
+            [
+              {
+                allowlist: [],
+                blocklist: [
+                  'QmUDBVyGwqKdSayk7kDKUaj9J41Ft1DWizcKUx5UmgMgGy',
+                  'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m',
+                  'example.com',
+                ],
+                fuzzylist: [],
+                name: 'first',
+                tolerance: 2,
+                version: 1,
+              },
+            ],
+            async ({ detector }) => {
+              const { result, type } = detector.check(
```

Comment:
It looks like we're only checking the `result` and `type` here, but the `check` method returns more properties in the object it returns. Specifically, it looks like the `version` property in the object will match the `version` that was passed in the config object. The reason why you're getting a coverage error is that you're not testing what happens when you don't pass a `version` property in the config object. Thoughts on testing that scenario?